builtin/providers/aws/resource_aws_ecr_repository.go

```diff
@@ -117,7 +119,31 @@ func resourceAwsEcrRepositoryDelete(d *schema.ResourceData, meta interface{}) er
 		return err
 	}
 
-	log.Printf("[DEBUG] repository %q deleted.", d.Get("arn").(string))
+	log.Printf("[DEBUG] Waiting for ECR Repository %q to be deleted", d.Id())
+	err = resource.Retry(20*time.Minute, func() *resource.RetryError {
+		_, err := conn.DescribeRepositories(&ecr.DescribeRepositoriesInput{
+			RepositoryNames: []*string{aws.String(d.Id())},
+		})
+
+		if err != nil {
+			awsErr, ok := err.(awserr.Error)
+			if !ok {
+				return resource.NonRetryableError(err)
+			}
+
+			if awsErr.Code() == "RepositoryNotFoundException" {
+				return nil
+			}
+
+			return resource.NonRetryableError(err)
+		}
+
+		return resource.RetryableError(
+			fmt.Errorf("%q: Timeout while waiting for the ECR Repository to be deleted", d.Id()))
+	})
```

Comment:
We don't check the `err` from `resource.Retry` here, so if the time limit hits we still remove from state, is that correct behavior?  If you hit any of the `resource.NonRetryableError` paths then we'll still remove from state because we don't check the error